app/controllers/api/v1/bookmarks_controller.rb

```diff
@@ -0,0 +1,117 @@
+class Api::V1::BookmarksController < Api::V1::BaseController
```

Comment:
With all my focus on the Open Doors use case, I hadn't thought of the two bookmark types possibly being used in future - I'll rename this and have a think about whether the URL end-point needs to be changed as well. (i.e. would it still make sense to POST both external and internal bookmarks to `api/v1/bookmarks`, or should there be a separate `api/v1/external-bookmarks`, and if so what happens to GET requests when we get around to them? Should both types be returned by a call to `bookmarks`?)